### ISSUE TITLE
Destination should not be SIGTERM-ed if source fails

### ISSUE THREAD

**Issue by edgao** [airbytehq/airbyte#17926]
## Current Behavior
If the source connector container fails (e.g. due to OOMKill), the destination connector container is SIGTERM-ed 60 seconds later. Example in https://cloud.airbyte.io/workspaces/94f992c9-be44-4e0f-8194-dfcac73b88a2/connections/6e1f6af6-cceb-4dd6-a2e7-c10f5529e983/status#696389::0:
* `2022-10-12 15:44:12 source > Killed`
* `2022-10-12 15:45:13 replication-orchestrator > (pod: jobs / destination-bigquery-write-696389-0-uhdzm) - Destroying Kube process.` (followed by a stacktrace `Suppressed: io.airbyte.workers.exception.WorkerException: Destination process exit with code 143. This warning is normal if the job was cancelled.`)

This SIGTERM prevented the destination from committing data to the `_raw` tables. This then caused normalization to also fail (because this was the first sync, and so the `_raw` tables didn't exist at all).

This also means we're not committing a new state message, so we're wasting effort. (it does mean that we're not losing data at least, since the next sync will just restart those streams from scratch / previous state)

## Expected Behavior
The destination connector should be allowed to run long enough to commit all the data the source produced, even if the source crashes.

**Comment by benmoriceau** [comment]
Yes should, There is some effort in changing when the state is persisted https://github.com/airbytehq/airbyte/issues/16762. I feel that the 2 stories are related. I'll push both issues in our grooming. (FYI @salima-airbyte @gosusnp )

Do we need the bandaid now. To be honest I have no idea about what the wait time should be and how do we want to configure it (per connection, global...)

**Comment by edgao** [comment]
connector pov: bumping the 60 sec wait time to something like 15 minutes globally would feel fine - we probably still have issues in connections with many streams, but the common case will work better. I don't know if having this configurable per-connection (or whatever) is super valuable, since presumably users would be incentivized to just set it arbitrarily high.

No idea if there are negative implications from that... does the "destination crash -> kill source" logic use the same timeout? (in which case we'd also want to keep the fast timeout, or ideally separate them)

**Comment by gosusnp** [comment]
Looking into how we currently kill processes, I don't think we have an explicit "if source dies, shutdown the destination".
Instead, I think our policy is if an exception is raised, shutdown the ExectorService which will in turn try to close any Connector within a 60s grace period.

The good news is that with our current code structure, we should be able to have a config that is specific for source/destinations. It should be possible to have it configured at the connection level but this doesn't really seem to make a lot of sense to me.

I'd probably want to hash-out quickly the downsides of this, but I think we can apply some quick band-aid here.
